src/webnn_native/ops/Unary.cpp

```diff
@@ -19,19 +19,24 @@
 
 namespace webnn_native { namespace op {
 
-    MaybeError Unary::Validate() {
-        MaybeError maybeError = OperatorBase::Validate();
+    MaybeError Unary::ValidateAndInferOutputInfo() {
+        MaybeError maybeError = OperatorBase::ValidateAndInferOutputInfo();
         if (maybeError.IsError()) {
             return maybeError;
         }
 
-        if (mOpType == UnaryOpType::kSoftmax) {
-            auto input = mInputs[0];
-            if (input->Rank() != 2) {
-                return DAWN_VALIDATION_ERROR("Input dimensions is incorrect.");
+        if (!mInputs.empty()) {
+            if (mOpType == UnaryOpType::kSoftmax) {
+                if (mInputs[0]->Shape().size() != 2) {
+                    return DAWN_VALIDATION_ERROR("Argument input of softmax is not a 2D tensor.");
+                }
+            } else {
+                if (mInputs[0]->Shape().empty()) {
```

Comment:
Softmax also doesn't support empty shape?

Reply:
Softmax input should be 2D:  "Argument input of softmax is not a 2D tensor." at [L30](https://github.com/webmachinelearning/webnn-native/pull/165/files#diff-aec0153e8103a89090f52837c74456a9ce32c8bafb72821141efd2258c0634a7R30). @fujunwei

Reply:
So why do you add the empty shape here?

Reply:
It is to check empty shape for other unary ops.